--- xmoto-0.5.10/src/xdgbasedir/src/basedir.c	2013-12-14 21:03:35.000000000 -0500
+++ xmoto-0.5.10/src/xdgbasedir/src/basedir.c	2013-12-14 21:10:24.000000000 -0500
@@ -25,6 +25,38 @@
 /** @file basedir.c
   * @brief Implementation of the XDG Base Directory specification. */
 
+#include <Carbon/Carbon.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <objc/objc.h>
+#include <NSSystemDirectories.h>
+
+CFArrayRef NSSearchPathForDirectoriesInDomains (NSSearchPathDirectory directory,
+   NSSearchPathDomainMask domainMask,
+   BOOL expandTilde
+);
+
+char *userdir(NSSearchPathDirectory d, CFStringRef subdir) {
+CFArrayRef a = NSSearchPathForDirectoriesInDomains(d, NSUserDomainMask, YES);
+CFStringRef s = CFArrayGetValueAtIndex(a, 0);
+   CFURLRef u = CFURLCreateWithFileSystemPath(NULL, s, kCFURLPOSIXPathStyle,
+       true);
+   CFRelease(a);
+
+   if (subdir) {
+       CFURLRef u2 = CFURLCreateCopyAppendingPathComponent(NULL, u, subdir,
+           true);
+       CFRelease(u);
+       u = u2;
+   }
+
+UInt8 path[PATH_MAX+1];
+   CFURLGetFileSystemRepresentation(u, true, path, sizeof(path));
+   CFRelease(u);
+
+return strdup((char*)path);
+}
+
+
 #if defined(HAVE_CONFIG_H) || defined(_DOXYGEN)
 #include <config.h>
 #endif
@@ -299,6 +331,11 @@
  */
 static int xdgUpdateHomeDirectories(xdgCachedData* cache)
 {
+	cache->dataHome = userdir(NSApplicationSupportDirectory, NULL);
+    cache->configHome = userdir(NSLibraryDirectory, CFSTR("Preferences"));
+	cache->cacheHome = userdir(NSCachesDirectory, NULL);
+    return TRUE;
+
 	const char* env;
 	char* home, *defVal;
 
